lib/lgy/service.rb

```diff
@@ -130,6 +130,11 @@ def get_coe_documents
       end
     end
 
+    # N.B. LGY does not validate that the requested document belongs to the
+    # user with the specified EDIPI. This may change in the future, but as of
+    # 2023-01-05, this is not the case. In the meantime, please use this method
+    # in conjunction with #get_coe_documents to ensure the requested document
+    # is within the list of documents possessed by the currently logged in user.
     def get_document(id)
```

Comment:
Or return 404 same as it does further down the line.

Reply:
To tack on to Kyle’s suggestion I might add that it might be worth considering moving the validation logic to the service instead of doing it in the controller

Reply:
Then you can raise a Common::Exceptions::RecordNotFound and the controller will handle the 404 stuff for you

Reply:
Yes that sounds best @jerekshoe, I agree.